tools/perfscale-load-generator/burst/burst.go

```diff
@@ -44,104 +44,115 @@ type BurstJob struct {
 	UUID       string
 	objType    string
 	done       <-chan time.Time
+	watchers   map[string]*watcher.ObjListWatcher
 }
 
 // NewBurstJob
-func newBurstJob(virtClient kubecli.KubevirtClient, workload *config.Workload, d <-chan time.Time) *BurstJob {
-	uid, _ := uuid.NewUUID()
+func newBurstJob(virtClient kubecli.KubevirtClient, workload *config.Workload, uuid string, d <-chan time.Time) *BurstJob {
 	return &BurstJob{
 		virtClient: virtClient,
 		Workload:   workload,
-		UUID:       uid.String(),
+		UUID:       uuid,
 		done:       d,
+		watchers:   map[string]*watcher.ObjListWatcher{},
 	}
 }
 
 func (b *BurstLoadGenerator) Delete(virtClient kubecli.KubevirtClient, workload *config.Workload) {
-	j := newBurstJob(virtClient, workload, b.Done)
+	j := newBurstJob(virtClient, workload, b.UUID, b.Done)
 	j.DeleteWorkload()
+	j.stopAllWatchers()
 }
 
 func (b *BurstLoadGenerator) Run(virtClient kubecli.KubevirtClient, workload *config.Workload) {
-	j := newBurstJob(virtClient, workload, b.Done)
+	j := newBurstJob(virtClient, workload, b.UUID, b.Done)
 	j.CreateWorkload()
+	// only stop watchers if the test will not delete the objs, otherwise the watchers will watch for the obj deletions
+	if !flags.Delete {
+		j.stopAllWatchers()
+	}
 }
 
 func (b *BurstJob) CreateWorkload() {
 	log.Log.V(1).Infof("Burst Load Generator CreateWorkload")
 
-	var wg sync.WaitGroup
-	obj := b.Workload.Object
-	for replica := 1; replica <= b.Workload.Count; replica++ {
-		select {
-		case <-b.done:
-			log.Log.V(1).Infof("Burst Load Generator duration has timed out")
-			return
-		default:
-		}
-
-		log.Log.V(2).Infof("Replica %d of %d", replica, b.Workload.Count)
-		newObject, err := utils.Create(b.virtClient, replica, obj, b.UUID)
+	// The watcher must be created before to be able to watch all events related to the objs.
+	// This is important because before creating each obj we need to create a obj watcher for each obj type.
+	objSpec := b.Workload.Object
+	objSample := renderObjSpecTemplate(objSpec, b.UUID)
+	b.createWatcherIfNotExist(objSample)
+
+	// Create all replicas
+	for r := 1; r <= b.Workload.Count; r++ {
+		log.Log.V(2).Infof("Replica %d of %d", r, b.Workload.Count)
+		idx := r
+		_, err := utils.CreateObjectReplica(b.virtClient, objSpec, &idx, b.UUID)
 		if err != nil {
 			continue
 		}
-		if b.objType == "" {
-			b.objType = objUtil.GetObjectResource(newObject)
-		}
+	}
 
-		wg.Add(1)
-		go func(newObject *unstructured.Unstructured) {
-			defer wg.Done()
-			b.watchCreate(newObject)
-			log.Log.Infof("obj %s is available", newObject.GroupVersionKind().Kind)
-		}(newObject)
+	// Wait all objects be Running
+	for objType, objWatcher := range b.watchers {
+		log.Log.Infof("Wait for obj(s) %s to be available", objType)
+		objWatcher.WaitRunning(b.Workload.Timeout.Duration)
```

Comment:
nit: shouldn't the timeout be a property of the watcher?

Reply:
Maybe because right now we have only one kind of timeout. But we could have different ones based on the obj type and test configuration....